### PR TITLE
[support] Remove some dead code

### PR DESCRIPTION
**Pull request by nsivov**
Both 'ptv' and 'tv' are unused since select() was replaced with poll() in 57b521e.

### CODE REVIEW

File: mono/metadata/threadpool-io.h [hunk 1]
```diff
@@ -15,6 +15,10 @@ typedef struct _MonoIOSelectorJob MonoIOSelectorJob;
 
 TYPED_HANDLE_DECL (MonoIOSelectorJob);
 
+ICALL_EXPORT
+void
+ves_icall_System_IOSelector_Add (gpointer handle, MonoIOSelectorJobHandle job_handle, MonoError* error);
```
Comment (CoffeeFlux) [inline mono/metadata/threadpool-io.h]: Icalls should be defined in the icall-def files. This particular function appears to already be defined there - do the signatures not match? I don't see any error when building on OSX at least.

File: support/signal.c [hunk 2]
```diff
@@ -481,13 +481,6 @@ wait_for_any (signal_info** signals, int count, int *currfd, struct pollfd* fd_s
...
 	// Once a second, stop to check if the VM is shutting down.
 	do {
-		struct timeval tv;
-		struct timeval *ptv = NULL;
-		if (timeout != -1) {
-			tv.tv_sec  = timeout / 1000;
-			tv.tv_usec = (timeout % 1000)*1000;
-			ptv = &tv;
-		}
 		r = poll (fd_structs, count, timeout);
```
Comment (jaykrell) [inline support/signal.c]: Is `poll` expecting `ptv`?

Comment (jaykrell) [reply support/signal.c]: Er, nevermind that, But the intent was to loop while `time remaining` and reduce it as it goes?

File: mono/metadata/threadpool-io.h (hunk 1 above)
Comment (nsivov) [reply mono/metadata/threadpool-io.h]: They do match, but apparently threadpool-io.c does not include icall-decl.h header like, I'll fix it up. I get this warning with gcc 9.2.1, that's probably not what you're using on osx. To clean it up one step further, should I remove ves_icall_System_IOSelector_Remove() aswell, moving it to icall-decl.h?

File: support/signal.c (hunk 2 above)
Comment (nsivov) [reply support/signal.c]: With select() it was called with same timeout every time, same as it is now with poll(). So I'm not sure what you're suggesting.

File: mono/metadata/threadpool-io.h (hunk 1 above)
Comment (jaykrell) [reply mono/metadata/threadpool-io.h]: No. remove is different, as I recall -- NOHANDLES vs. HANDLES.

Comment (CoffeeFlux) [reply mono/metadata/threadpool-io.h]: I think you can move the NOHANDLES ones as well? Not sure what the convention is though.
Looking further, the warning also appears to show up on netcore, but in that case I think the implementation needs to be ifdef'd out? Not sure what IOSelector is, but it looks like some mono-specific legacy thing.
```
threadpool-io.c:603:1: warning: no previous prototype for function 'ves_icall_System_IOSelector_Add' [-Wmissing-prototypes]
ves_icall_System_IOSelector_Add (gpointer handle, MonoIOSelectorJobHandle job_handle, MonoError* error)
```

Comment (nsivov) [reply mono/metadata/threadpool-io.h]: Just to make sure I understand, do you want me to move _Remove() to icall-decl.h? (out of threadpool-io.h is the point, I think)

**Comment by nsivov** [comment]
Thanks for taking care of this @akoeplinger

**Pull request merged**